File: backend-v5/lms/forum/ForumPost.php
<?php

// Modul: lms/forum
// Klasa: ForumPost
// Opis: jedna poruka na forumu

require_once(Config::$backend_path."core/DB.php");
require_once(Config::$backend_path."core/Person.php");

// FIXME: u bazi ne postoji tabela forum
// polje forum tabele topic se zove "projekat"


class ForumPost {
	public $id;
	public $subject, $time, $authorId, $topicId, $text;
	public $author;

	public static function fromId($id) {
		$q10 = DB::query("select p.naslov, UNIX_TIMESTAMP(p.vrijeme), p.osoba, p.tema, pt.tekst, o.ime, o.prezime, o.brindexa, a.login from bb_post as p, bb_post_text as pt, osoba as o, auth as a where p.id=$id and pt.post=$id and p.osoba=o.id and o.id=a.id");
		if (mysql_num_rows($q10)<1) {
			throw new Exception("unknown forum post");
		}
		
		$fp = new ForumPost;
		$fp->id = $id;
		$fp->subject = mysql_result($q10,0,0);
		$fp->time = mysql_result($q10,0,1);
		$fp->authorId = mysql_result($q10,0,2);
		$fp->topicId = mysql_result($q10,0,3);
		$fp->text = mysql_result($q10,0,4);

		$fp->author = new Person;
		$fp->author->id = $fp->authorId;
		$fp->author->name = mysql_result($q10,0,5);
		$fp->author->surname = mysql_result($q10,0,6);
		$fp->author->studentIdNr = mysql_result($q10,0,7);
		$fp->author->login = mysql_result($q10,0,8);
		
		return $fp;
	}

	public function delete() {
		$q10 = DB::query("delete from bb_post where id=".$this->id);
		$q20 = DB::query("delete from bb_post_text where post=".$this->id);
	}

	public function update() {
		$q10 = DB::query("update bb_post set naslov='".$this->subject."' where id=".$this->id);
		$q20 = DB::query("update bb_post_text set tekst='".$this->text."' where post=".$this->id);

	}
}

?>

File: rorbackend/app/models/lms/forum/forum_post.rb
class Lms::Forum::ForumPost < ActiveRecord::Base
  # Uncomment following lines if working with legacy database
  # set_table_name 'bb_post'
  # set_primary_key :id
  # alias_attribute :id, :id
  # alias_attribute :subject, :naslov
  # alias_attribute :time, :vrijeme
  # alias_attribute :author_id, :osoba
  # alias_attribute :forum_topic_id, :tema

  # Uncomment following lines if working with legacy database
  # TABLE_NAME = 'bb_post'
  # ID = TABLE_NAME + '.' + 'id'
  # SUBJECT =  TABLE_NAME + '.' + 'naslov'
  # TIME =  TABLE_NAME + '.' + 'vrijeme'
  # AUTHOR_ID =  TABLE_NAME + '.' + 'osoba'
  # FORUM_TOPIC_ID =  TABLE_NAME + '.' + 'tema'
 
  # Comment following lines if working with legacy database
  TABLE_NAME = 'lms_forum_forum_posts'
  ID = TABLE_NAME + '.' + 'id'
  SUBJECT =  TABLE_NAME + '.' + 'subject'
  TIME =  TABLE_NAME + '.' + 'time'
  AUTHOR_ID =  TABLE_NAME + '.' + 'author_id'
  FORUM_TOPIC_ID =  TABLE_NAME + '.' + 'forum_topic_id'

  ALL_COLUMNS = [ID, SUBJECT, TIME, AUTHOR_ID, FORUM_TOPIC_ID]
  
  
  belongs_to :author, :class_name => "Core::Person"
  belongs_to :topic
  has_one :forum_post_text
  
  def self.from_id(id)
    select_columns = [(Lms::Forum::ForumPost).SUBJECT, (Lms::Forum::ForumPost).TIME, (Lms::Forum::ForumPost).AUTHOR_ID, (Lms::Forum::ForumPost).TOPIC_ID, (Lms::Forum::ForumPost).TEXT, (Core::Person).ID, (Core::Person).NAME, (Core::Person).SURNAME, (Core::Person).STUDENT_ID_NUMBER, (Core::Auth).EMAIL]
    post = (Lms::Forum::ForumPost).joins(:author => :auth).joins(:forum_post_text).where(:id => id).select(select_columns)
    
    return post
  end
  
  
  def self.delete_o(id)
    (Lms::Forum::ForumPost).transaction do
      (Lms::Forum::ForumPostText).transaction do
        num_returned = 0
        num_returned += (Lms::Forum::ForumPost).delete(id)
        num_returned += (Lms::Forum::ForumPostText).delete_all(:forum_post_id => id)
        raise ActiveRecord::RecordNotFound if (num_returned == 0)
      end
    end
    return true
  end
  
  def update_o(id, subject, text)
    (Lms::Forum::ForumPost).transaction do
      (Lms::Forum::ForumPostText).transaction do
        forum_post = (Lms::Forum::ForumPost).find(id)
        forum_post_text = (Lms::Forum::ForumPostText).where(:forum_post_id => id).first
        forum_post.subject = subject
        forum_post_text = text
        
        return (forum_post.save and forum.post_text.save)
      end
    end
  end
end
